cpp/test/Ice/idleTimeout/AllTests.cpp

```diff
@@ -0,0 +1,74 @@
+//
+// Copyright (c) ZeroC, Inc. All rights reserved.
+//
+
+#include "Test.h"
+#include "TestHelper.h"
+
+using namespace std;
+using namespace Ice;
+using namespace Test;
+
+// The client and server have the same idle timeout (1s) and the server has EnableIdleCheck = 1.
+// We verify that the server's idle check does not abort the connection as long as this connection receives heartbeats,
+// even when the heartbeats are not read off the connection in a timely manner.
+// To verify this situation, we use an OA with a 1-thread thread pool and use this unique thread for a long synchronous
+// dispatch (sleep).
+void
+testIdleCheckDoesNotAbortConnectionWhenThreadPoolIsExhausted(const TestIntfPrx& p)
+{
+    cout << "testing that the idle check does not abort a connection that receives heartbeats... " << flush;
```

Comment:
It could be less time sensitive to not rely on heartbeats but instead make an asynchronous sleep followed by an ice_ping.

Reply:
I don't understand what you're suggesting.

> not rely on heartbeats 

The whole point of this test is to make sure a connection is not idle-check-aborted because it receives heartbeats.

Reply:
Ok